system/lib/compiler-rt/lib/asan/asan_poisoning.cc

```diff
@@ -188,6 +190,10 @@ uptr __asan_region_is_poisoned(uptr beg, uptr size) {
     if (!AddrIsInMem(beg) && !AddrIsInShadow(beg)) return 0;
     if (!AddrIsInMem(end) && !AddrIsInShadow(end)) return 0;
   } else {
+#if SANITIZER_EMSCRIPTEN
+    // Null pointer handling, since Emscripten does not crash on null pointer
+    if (!beg) return 1;
```

Comment:
Should this include the shadow shadow memory?

Reply:
This is the function that returns the first inaccessible memory address, and it also returns `nullptr` for the entire region not being poisoned, so if we pass `nullptr` in, the caller will think the region is accessible. Hence, a hack to make it return 1 instead of 0 so it gets reported. I'll update the comments.